main_routers/workshop_router.py

```
@@ -1215,6 +1552,178 @@ async def list_audio_files(directory: str):
+            logger.warning(f"检测到非法文件名尝试: {chara_file_name}")
+            return JSONResponse({
+                "success": False,
+                "error": "非法文件名"
+            }, status_code=400)
+        
+        # 如果没有传递 character_card_name，尝试从文件名提取
+        if not character_card_name and safe_chara_name:
+            if safe_chara_name.endswith('.chara.json'):
+                character_card_name = safe_chara_name[:-11]  # 去掉 .chara.json 后缀
+        
+        # 检查是否已存在workshop_meta.json文件（防止重复上传）
```

Ruff (0.14.10)

Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)